### PR TITLE
Update to latest almanac structs and functions

### PR DESCRIPTION
**Pull request by ljbade**
Requires https://github.com/swift-nav/libswiftnav/pull/269

### CODE REVIEW

**Comment by ljbade** [comment]
Ready for review @swift-nav/estimation

**Comment by ljbade** [comment]
Also @swift-nav/firmware can review

**Comment by ljbade** [comment]
So far HITL tests indicate slightly lower performance, but that doesn't make sense as HITL doesn't touch almanac or the new ephemeris functions. Existing ephemeris functions haven't changed except for the change to `fit_interval` units, but a problem there would result in no solutions at all.

I think I need to wait for more HITL runs to get through the backlog.

**Comment by ljbade** [comment]
Second HITL looked better, a few things I need to look at though.

**Comment by ljbade** [comment]
Their is an increase in RTK loss metrics due to sequences like:
```
INFO Software reset detected.
INFO Reset reason: 14
INFO Piksi Starting...
INFO Firmware Version: v0.21-130-g923c808-dirty
INFO Built: Apr  5 2016 14:05:08
INFO No telemetry radio found on UARTA, skipping configuration.
INFO No telemetry radio found on UARTB, skipping configuration.
INFO NAP firmware version: v0.16
INFO Antenna changed to: External
INFO Loaded last position solution from file: 37.7625 -122.3891 -4.4
INFO Time set to: Tue Apr  5 08:42:02 2016
 (quality=1)
INFO GPS L1CA 18 synced
INFO GPS L1CA 26 synced
INFO GPS L1CA 5 synced
INFO GPS L1CA 4 synced
INFO GPS L1CA 29 synced
INFO GPS L1CA 21 synced
INFO GPS L1CA 13 synced
INFO GPS L1CA 20 synced
INFO GPS L1CA 15 synced
INFO GPS L1CA 16 synced
INFO GPS L1CA 29 New untrusted ephemeris
INFO GPS L1CA 21 New untrusted ephemeris
WARN GPS L1CA 4 Latest ephemeris is unhealthy. Ignoring satellite.
INFO GPS L1CA 4 New untrusted ephemeris
INFO GPS L1CA 26 New untrusted ephemeris
INFO GPS L1CA 5 New untrusted ephemeris
INFO GPS L1CA 4 unhealthy, dropping
INFO GPS L1CA 4 New ephemeris candidate
INFO GPS L1CA 5 New trusted ephemeris
INFO GPS L1CA 15 New untrusted ephemeris
INFO GPS L1CA 16 New untrusted ephemeris
INFO GPS L1CA 18 New untrusted ephemeris
INFO GPS L1CA 20 New untrusted ephemeris
INFO GPS L1CA 21 New trusted ephemeris
WARN dgnss_baseline returned error: -1
INFO GPS L1CA 26 New trusted ephemeris
WARN dgnss_baseline returned error: -1
WARN dgnss_baseline returned error: -1
INFO GPS L1CA 29 New trusted ephemeris
WARN dgnss_baseline returned error: -1
WARN dgnss_baseline returned error: -1
WARN dgnss_baseline returned error: -1
WARN dgnss_baseline returned error: -1
WARN dgnss_baseline returned error: -1
INFO Initializing DGNSS filters
WARN dgnss_baseline returned error: -1
WARN dgnss_baseline returned error: -1
WARN dgnss_update. baseline estimate error: -3
```
During startup

**Comment by ljbade** [comment]
Considering that the HITL tests for https://github.com/swift-nav/piksi_firmware/pull/632 also have the same problems, and is currently in master, it appears OK to ignore the RTK losses during startup.

Also I remember seeing the same behaviour on Piksi since the end of last year.

If anything the HITL test needs to be tweaked to ignore these messages during startup, or firmware should suppress them.

@denniszollo What is your opinion?

**Comment by ljbade** [comment]
Ready for review @swift-nav/firmware 

I think I can ignore the HITL issues since they do not come from changes in this PR.

File: src/manage.c [hunk 1]
```diff
@@ -268,13 +268,14 @@ static u16 manage_warm_start(gnss_signal_t sid, const gps_time_t* t,
     } else {
       const almanac_t *a = &almanac[sid_to_global_index(sid)];
       if (a->valid) {
-        calc_sat_az_el_almanac(a, t->tow, t->wn-1024,
```
Comment (jacobmcnamee) [inline src/manage.c]: What's was the `t->wn-1024` supposed to do?

Comment (ljbade) [inline src/manage.c]: It appeared to be a crude way to handle GPS rollover. Currently GPS is in the second cycle of GPS weeks, thus the "true" WN is some number greater than 1024, but the over the wire WN in GPS nav data is mod 1024.

The ephemeris functions correctly handle this, it appeared the old almanac functions did not.

**Comment by jacobmcnamee** [comment]
lgtm assuming hitl is fine.

**Comment by ljbade** [comment]
@jacobmcnamee increase ram4 made Travis pass.

**Comment by ljbade** [comment]
@jacobmcnamee I locally tested a suggestion from @rge-exafore on the memory issue. It seems to work, so I'm doing more HITL testing.

In the meantime can you review the change to the stack locations.

**Comment by jacobmcnamee** [comment]
lgtm

**Pull request merged**